### ISSUE TITLE
ATS tries to DNS resolve host header

### ISSUE THREAD

**Issue by amoghyermalkar123** [apache/trafficserver#8553]
I currently have a ATS cache-hierarchy system in place. I have multiple child ATS nodes and multiple parent ATS nodes. So for example, if i get a request on a child node, it first DNS resolves its parents defined in parent.config. If the DNS fails for some reason, or the parents are unreachable it then tries to DNS resolve the host header of the request. I want to avoid doing that. The reason for this is just my business requirement.

What I have done:
I have searched for configurations, read documents on hostdb module, but I have not found a configuration that allows me to do that.

I basically want to fail the request if parents fail, but want to avoid DNS resolving the host header. This is also because for further requests on the same mapping, ATS subsequently keeps resolving the host header and never tries resolving parents.

Please help me in understanding the DNS algorithm for parent servers that ATS follows and any best practises or configurations i can follow. 

Thanks in advance.

**Comment by amoghyermalkar123** [comment]
It'd be amazing if I could get a state machine as well :)

**Comment by amoghyermalkar123** [comment]
Hi, could I please get some help :)

**Comment by amoghyermalkar123** [comment]
Hi, is there some additional information required on this issue? Is this ready to be assigned?

**Comment by jvgutierrez** [comment]
have you tried `proxy.config.http.no_dns_just_forward_to_parent`? It's documented on https://docs.trafficserver.apache.org/en/latest/admin-guide/files/records.config.en.html#proxy-config-http-no-dns-just-forward-to-parent. We are successfully using it.

**Comment by amoghyermalkar123** [comment]
Thanks, I will try this out. 
Is there a FSM diagram available anywhere for any kind of algorithms in ATS? Be it DNS lookup or parent selection, I have a lot of missing dots in my head that i am trying to connect about various algorithms. 

I saw the debug logs where state transitions are mentioned, I was able to make a decent FSM from those by testing my ATS nodes under a bunch of different scenarios, but I feel this is a good chance for me to ask for one.

All in all, thanks for the linked document.
